test/functional/lib/redirect-handler/blockOrObserve.test.ts

```diff
@@ -184,6 +183,30 @@ describe('lib/redirect-handler/blockOrObserve', () => {
       })
     })
 
+    it('Should update tab from originUrl to redirectUrl for the first time', async () => {
+      await addRuleToDynamicRuleSet({
+        originUrl: 'https://ipfs.io/ipns/en.wikipedia-on-ipfs.org',
+        redirectUrl: 'http://localhost:8080/ipns/en.wikipedia-on-ipfs.org'
+      })
+      expect(browserMock.tabs.query.called).to.be.true
+      expect(browserMock.tabs.update.called).to.be.true
+      expect(browserMock.tabs.update.lastCall.args).to.deep.equal([40, { url: 'http://localhost:8080/ipns/en.wikipedia-on-ipfs.org' }])
+    })
+
+    it('Should wait for the tab to exist before updating tab from originUrl to redirectUrl for the first time', async () => {
+      const clock = sinon.useFakeTimers();
+      browserMock.tabs.query.onCall(0).resolves([])
+      browserMock.tabs.query.onCall(1).resolves([{ id: 40 }])
+      await addRuleToDynamicRuleSet({
+        originUrl: 'https://ipfs.io/ipns/en.wikipedia-on-ipfs.org',
+        redirectUrl: 'http://localhost:8080/ipns/en.wikipedia-on-ipfs.org'
+      })
+      await clock.tickAsync(400)
```

Comment:
nit: could we base this on `MAX_RETRIES_TO_UPDATE_TAB` somehow, so the reader can figure out why we sleep here?